framework/core/swapchain.h

```diff
@@ -159,5 +185,9 @@ class Swapchain
 	    {VK_FORMAT_B8G8R8A8_SRGB, VK_COLOR_SPACE_SRGB_NONLINEAR_KHR}};
 
 	std::set<VkImageUsageFlagBits> image_usage_flags;
+
+	VkImageCompressionFlagsEXT requested_compression{VK_IMAGE_COMPRESSION_DEFAULT_EXT};
+
+	VkImageCompressionFixedRateFlagsEXT requested_compression_fixed_rate{VK_IMAGE_COMPRESSION_FIXED_RATE_NONE_EXT};
```

Comment:
These seem like they'd belong in the `SwapchainProperties` class, rather than in the `SwapChain` class itself.  All the other properties that are configurable at runtime are there (excepting some weirdness with the `image_usage_flags`).

Reply:
Also, migrating these into the SwapchainProperties should make resolving any conflicts with #973 easier to handle, I think.  

Feel free to move #973 into draft mode and/or comment on it if you want it to be deferred until after this is merged.

Reply:
yes please, I'll address your comments but I would appreciate if you could then adapt this in #973 🙏

Reply:
Sure thing.